Code Blocks Guidance (#1903)
grommet/hpe-design-system: aries-site/src/pages/templates/code-blocks.mdx

<Example
  code={[
    "https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/code-blocks/CodeBlocks.js",
    "https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/code-blocks/DarkCodeTheme.js",
    "https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/code-blocks/LightCodeTheme.js",
  ]}
  showResponsiveControls={['laptop', 'mobile']}
  template
>
  <CodeBlockExample />
</Example>

## Guidance

Code blocks should be created using the [react-syntax-highlighter](https://github.com/react-syntax-highlighter/react-syntax-highlighter) tool.

Prism or Highlighter.js can be used with react-syntax-highlighter with Highlighter.js being the default. Highlighter.js occasionally has trouble highlighting jsx so it is recommended to use Prism instead. 
The documentation for using Prism with react-syntax-highlighter includes a [list of available languages](https://github.com/react-syntax-highlighter/react-syntax-highlighter/blob/master/AVAILABLE_LANGUAGES_PRISM.MD).

react-syntax-highlighter can be installed with the following command:

> 

Reviewer: 👍

`yarn add react-syntax-highlighter`

Import the react-syntax-highlighter specifying Prism as the highlighter with the following:

`import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';`

### Theming

The following themes should be used to style code blocks.

> [Light mode theme](https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/code-blocks/DarkCodeTheme.js)

Reviewer: are we going to change these links to point to the new theme files or are we leaving these files in here as well?

Reviewer: sounds good! Looks Good other than that!!

import { Example } from '../../layouts';
import {
  CodeBlockExample
} from '../../examples';

<Example
  code={[
    "https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/code-blocks/CodeBlocks.js",
    "https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/code-blocks/DarkCodeTheme.js",
    "https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/code-blocks/LightCodeTheme.js",
  ]}
  showResponsiveControls={['laptop', 'mobile']}
  template
>
  <CodeBlockExample />
</Example>

## Guidance

> Code blocks should be created using the [react-syntax-highlighter](https://github.com/react-syntax-highlighter/react-syntax-highlighter) tool.

Reviewer 2: Softening just a bit to allow for non-React projects.

```suggestion
Code blocks can be created using the [react-syntax-highlighter](https://github.com/react-syntax-highlighter/react-syntax-highlighter) tool.
```

> Prism or Highlighter.js can be used with react-syntax-highlighter with Highlighter.js being the default. Highlighter.js occasionally has trouble highlighting jsx so it is recommended to use Prism instead. 

Reviewer 2: ```suggestion
Prism or Highlighter.js can be used with react-syntax-highlighter. Highlighter.js is the default, however it can have trouble highlighting JSX, so it is recommended to use Prism instead. 
```

> The documentation for using Prism with react-syntax-highlighter includes a [list of available languages](https://github.com/react-syntax-highlighter/react-syntax-highlighter/blob/master/AVAILABLE_LANGUAGES_PRISM.MD).

Reviewer 2: ```suggestion
Prism with react-syntax-highlighter support syntax highlighting for [hundreds of languages](https://github.com/react-syntax-highlighter/react-syntax-highlighter/blob/master/AVAILABLE_LANGUAGES_PRISM.MD).
```

react-syntax-highlighter can be installed with the following command:

For yarn users: `yarn add react-syntax-highlighter`

> For npm users: `npm install react-syntax-highlighter --save`

Reviewer 2: Should we modify our [code Markdown styling](https://github.com/grommet/hpe-design-system/blob/77d3a3e7d035db81b37a97a6c970132f9eac0dd3/aries-site/src/components/content/MarkdownComponents.js#L32) so that it is wrapped in `<SyntaxHighlighter />`?

Could be nice :)

Reviewer 2: I'm good with that.
